python-checks/src/main/java/org/sonar/python/checks/tests/AssertAfterRaiseCheck.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.tests;
+
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.IssueLocation;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.ExpressionStatement;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.Statement;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.plugins.python.api.tree.WithStatement;
+import org.sonar.python.tests.PytestUtils;
+import org.sonar.python.tests.UnittestUtils;
+
+@Rule(key = "S5915")
+public class AssertAfterRaiseCheck extends PythonSubscriptionCheck {
+  private static final String MESSAGE_MULTIPLE_STATEMENT = "Don’t perform an assertion here; An exception is expected to be raised before its execution.";
+  private static final String MESSAGE_SINGLE_STATEMENT = "Refactor this test; if this assertion’s argument raises an exception, the assertion will never get executed.";
+  private static final String MESSAGE_SECONDARY = null;
```

Comment:
I'm not sure it's a good practice to have secondary locations with null message. If none has been defined in the RSPEC, maybe we can create one.
What about `An exception is expected to be raised in this block.` ?

Reply:
That's what I was thinking, I saw that all methods about adding secondary had specifically "Nullable" but I was not sure that was a good Idea.
Indeed there is nothing in RSPEC, the message you provided sounds good to me !